src/cdk/scrolling/virtual-scroll-viewport.ts

```diff
@@ -202,6 +204,17 @@ export class CdkVirtualScrollViewport extends CdkVirtualScrollable implements On
       this.elementRef.nativeElement.classList.add('cdk-virtual-scrollable');
       this.scrollable = this;
     }
+
+    const ref = effect(
+      () => {
+        if (!this._changeDetectionNeeded()) {
+          return;
+        }
+        this._doChangeDetection();
+      },
+      {injector: inject(ApplicationRef).injector},
```

Comment:
Couldn't we let it resolve the injector automatically or does it have to be the app ref specifically? If it's the latter, we should have a comment.